### PR TITLE
docs: fail string std assertion

### PR DESCRIPTION
**Pull request by shawnharmsen**
closes: https://github.com/foundry-rs/book/issues/316
closes: https://github.com/foundry-rs/book/issues/226

Will add the remaining Std assertions and set to ready for review

### CODE REVIEW

File: src/reference/forge-std/assertApproxEqRel.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Asserts `a` is approximately equal to `b` with delta in percentage (An 18 decimal fixed point number, where 1e18 == 100%).
+
+### Examples
+
+```solidity
+uint256 a = 100;
+uint256 b = 200;
+function testFail () external {
+    assertApproxEqRel(a, b, 400000000000000000);
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/assertApproxEqRel.md:31]:
```suggestion
    assertApproxEqRel(a, b, 0.4e18);
```

Comment (ZeroEkkusu) [inline src/reference/forge-std/assertApproxEqRel.md:31]: This is cleaner - gives a better example of it should be used.

```suggestion
    assertApproxEqRel(a, b, 0.4e18);
```

File: src/reference/forge-std/fail.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+### Description
+
+Provides a clean and readable way to fail tests if a certain branch or execution point is hit.
+
+### Examples
+
+```solidity
+function test() external {
+    fail("failed");
+}
```
Comment (ZeroEkkusu) [inline src/reference/forge-std/fail.md:18]: Better example.

```suggestion
function test() external {
    for(int id; id < supply; ++i){
        if(collectible.ownerOf(id) == address(this)) return;
    }
    fail("Does not own any collectible.");
}
```

Comment (ZeroEkkusu) [inline src/reference/forge-std/fail.md:18]: Better example.

```suggestion
function testOwned() external {
    for(int id; id < supply; ++i){
        if(collectible.ownerOf(id) == address(this)) return;
    }
    fail("Does not own a collectible.");
}
```

**Review by ZeroEkkusu — COMMENTED** [review]
Could you also add it here as well:

- [Forge Standard Library Overview](https://book.getfoundry.sh/forge/forge-std.html#std-math)
- [Forge Standard Library Reference](https://book.getfoundry.sh/reference/forge-std/index.html#forge-stds-test)

File: src/reference/forge-std/assertEq.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+```
+
+```solidity
+function assertEq(bytes memory a, bytes memory b, string memory err) internal;
+```
+
+
+### Description
+
+Assert `a` is equal to `b`.
```
Comment (0xvv) [inline src/reference/forge-std/assertEq.md:24]: Asserts *

**Review by 0xvv — COMMENTED** [review]
nitpicking

**Comment by shawnharmsen** [comment]
I think I got everything

**Review by ZeroEkkusu — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]

**Pull request merged**